MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoPresenter.swift

```diff
@@ -4,23 +4,54 @@ final class HomePageTwoPresenter {
     private let useCase: MoviesUseCaseProtocol!
     private let router: AppRouter!
     
+    var categories: [CategoryViewModel]!
+    
     init (useCase: MoviesUseCaseProtocol, router: AppRouter) {
         self.useCase = useCase
         self.router = router
+        
+        categories = []
     }
     
     func setDelegate(delegate: HomePageTwoDelegate) {
         self.delegate = delegate
     }
     
+    func getAllCategories() {
+        getPopularMovies()
+        getTrendingMovies()
+        getTopRatedMovies()
+    }
+    
+    func getMovies(for subcategory: LocalSubcategory) -> [MovieViewModel] {
+        var movies: [MovieViewModel] = []
+        categories.forEach {
+            if $0.subcategoryMovies.keys.contains(subcategory) {
```

Comment:
Why do you need this check?
Having only `movies = $0.subcategoryMovies[subcategory] ?? []` should be enough.

Reply:
When I remove if statement movies don't present in the collection view.

![simulator_screenshot_F43ED74B-D606-46D0-B017-3FA64AC5955B](https://user-images.githubusercontent.com/82365531/127861597-e95e4e8c-2088-4da2-a98b-7078c229e448.png)